### ISSUE TITLE
[Question] State of Firestore C++ SDK for Desktop?

### ISSUE THREAD

**Issue by ScottPierce** [firebase/firebase-cpp-sdk#442]
### [REQUIRED] Please fill in the following fields:
  * Main Firebase Components in concern: Firestore + Auth
  * Platform you are using the C++ SDK on: Mac and Windows
  * Platform you are targeting: Desktop

### [REQUIRED] Please describe the question here:
I notice on the readme that there is no reference to Firestore C++. I found some old issues from 2019 that talk about firestore being in alpha.

I'm interested in using the Firestore C++ SDK in a Windows and Mac desktop application. What is the state of firestore currently? Is desktop supported for Windows and Mac environments, or would I be treading into alpha/beta territory?

**Comment by rafikhan** [comment]
@ScottPierce - Thanks for pointing out that the top level readme does not mention Firestore. That's a miss on our side and I filed an internal bug for it [b/190075287](b/190075287).

As for desktop support it is intended to make the local workflow easier but is not intended for publicly shipping code (hence the beta label). Given that it is open source you are welcome to try and ship it but do so at your own risk. If you make pull requests with bug fixes or enhancements we would take a look at it but platform specific issues are unlikely to get attention.

**Comment by ScottPierce** [comment]
@rafikhan What's the purpose behind the C++ library?

**Comment by rafikhan** [comment]
The overly simplified answer is that it made it easier to target multiple platforms with a single codebase. We've only tested it on the platforms where we've had developer demand (mobile platforms and the Unity Editor); however, if there was more demand for desktop support or other platforms we would seriously consider it.

**Comment by ttgil** [comment]
In what way would developer demand be expressed for the Firebase team to measure?  I've been building new products on the Firebase C++ SDK for a couple of years under the understanding that desktop would become a fully-supported platform that would eventually graduate out of beta and be used for shipping software.  Using Firebase together with Qt/C++ in cross-platform desktop and mobile apps is a great use case, as it is with Flutter.

**Comment by rafikhan** [comment]
@ttgil - We monitor a **lot** of channels, look at metrics, and speak with customers directly on an ongoing basis. One of the formal channels we look at is [here](https://firebase.google.com/support/troubleshooter/report/features); however, for your request I suggest voting on this issue. If you know others in the C++ community that want to leverage Firestore I would encourage you to have them do the same.

**Comment by ttgil** [comment]
> @ttgil - We monitor a **lot** of channels, look at metrics, and speak with customers directly on an ongoing basis. One of the many formal channels we look at is [here](https://firebase.google.com/support/troubleshooter/report/features); however, for your request I suggest voting on this issue. If you know others in the C++ desktop development community that want to leverage Firestore I would encourage you to have them do the same.

Why would someone fill out a feature request for a feature that is already implemented as a beta, as Firestore on desktop is?  I don't see a voting mechanism here.  

Is there some chance that the desktop implementations will NOT be supported for shipping software?  I've been working with it for 2 years, and have gotten assurances from members of the Firebase dev team that the desktop support will graduate from beta eventually.  So, what's changed?

**Comment by rafikhan** [comment]
> I don't see a voting mechanism here.

You can press the smiley face on the upper right hand corner of the first (or any relevant) comment in the thread and pick an appropriate emoji. We keep an eye on these.

> Is there some chance that the desktop implementations will NOT be supported for shipping software? I've been working with it for 2 years, and have gotten assurances from members of the Firebase dev team that the desktop support will graduate from beta eventually. So, what's changed?

When the Firestore C++ SDK graduates out of beta we will clearly document which platforms are supported for which use case. We try to stay consistent with other Firebase SDKs which at this time is mobile platforms intended for games and unity editor for desktop. As I mentioned before we regularly re-evaluate these based on customer demand and I encourage you voice that in the ways I mentioned above. I cannot make commitments or comment past that.

I'm not sure where you received assurances from our team on desktop support outside of the Unity Editor but saw you are on a thread with other googlers on this topic. We can continue this aspect of the conversation there.

**Comment by chkuang-g** [comment]
@ScottPierce  @ttgil 

Thank you for reporting your concerns.

If I read this correctly, the general concerns is that Firestore or Firebase C++ SDK in general for desktop is not out of beta. I would recommend to change this thread to a feature request by:
- Change the title to something like "FR: Firestore C++ SDK for desktop out of beta".
- Upvote the first post of this thread.

This can help us to understand the demand and to prioritize our resources.

Let us know if this works for you.
Shawn

**Comment by ScottPierce** [comment]
Sure

**Comment by natasha1007** [comment]
спасибо большое за внимание и понимание. 😍

**Comment by cedvdb** [comment]
The flutter demand for firebase support on desktop is growing and has more upvotes than this issue, so I'm not sure the smiley voting mechanism is accurate

One member of the community is trying to create a desktop variant, but honestly I don't see how that's remotely realistic.

**Comment by dconeybe** [comment]
Thanks you @cedvdb for your request for official desktop support. We've noted this but have no plans at this point for it. If this changes in the future we will for sure reply here.

**Comment by infiniteloopunroll** [comment]
I think that many active FB users would highly appreciate official support for the desktop C++ SDK. If there is anything that we can do to upvote this, beyond letting you know, would be great.

**Comment by JarrylWirth** [comment]
We would like to add our upvote for post-beta official support for the desktop C++ SDK.

Thank you.

**Comment by hwkim1127** [comment]
is there any updates on this issue?

**Comment by dconeybe** [comment]
Unfortunately, no updates here. The previous comment https://github.com/firebase/firebase-cpp-sdk/issues/442#issuecomment-1208784614 still applies (i.e. we've noted this request but have no plans at this point for it).

**Comment by cbenhagen** [comment]
@dconeybe are you aware of the official flutter plugins usage of the C++ SDK (https://github.com/firebase/flutterfire/pull/10496)? If not, could this influence the plans to take this SDK out of beta?

**Comment by dconeybe** [comment]
> @dconeybe are you aware of the official flutter plugins usage of the C++ SDK ([firebase/flutterfire#10496](https://github.com/firebase/flutterfire/pull/10496))? If not, could this influence the plans to take this SDK out of beta?

Yes, we are aware of the work going on in Flutter but cannot share any plans at this point about the C++ SDK going out of beta on the desktop. Someone will certainly update here if/when we have something concrete to announce.

**Comment by cbenhagen** [comment]
@everyone please show your support for the Firebase C++ SDK on Desktop by upvoting [this idea](https://firebase.uservoice.com/forums/948424-general/suggestions/46561888-desktop-windows-linux-and-macos-support-for-c). Thanks!
